tools/noopfactory/src/main/kotlin/com/datadog/tools/noopfactory/NoOpFactoryAnnotationProcessor.kt

```diff
@@ -268,11 +275,20 @@ class NoOpFactoryAnnotationProcessor : AbstractProcessor() {
     private fun generateFunctionReturnStatement(
         funSpecBuilder: FunSpec.Builder,
         returnType: TypeMirror,
+        functionParams: List<VariableElement>,
         isReturnNullable: Boolean
     ) {
+
         val type = returnType.asKotlinTypeName().copy(isReturnNullable)
         funSpecBuilder.returns(type)
-
+        if (!isReturnNullable && functionParams.isNotEmpty()) {
```

Comment:
why do we need this block? so let's say we have the following signature:

```
fun foobar(arg: String): String
```

Then it will add do `return arg`? and it will be essentially `no-op`. Because without this block we would try to create a new object, so that `input != output`. Am I right?

Reply:
yes...I need it to actually return the `arg` instead of creating a new string for return.